src/components/TabletApp/RightSide/QuickBook/QuickBook.tsx

```diff
@@ -1,81 +1,127 @@
 import React from "react";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import dayjs from "dayjs";
 import {
     Box,
     Button,
     Typography,
     Grid,
-    Paper,
     TextField,
     Autocomplete,
 } from "@mui/material";
 import EditCalendarIcon from "@mui/icons-material/EditCalendar";
-import { useState, useEffect } from "react";
-import axios from "axios";
-import { styled } from "@mui/material/styles";
-import { getParticipants } from "../../../../api/getRequests";
-import COLORS from "../../../../constants/CustomColors";
-
-//To be removed when globla theme is done
-const Item = styled(Paper)(({ theme }) => ({
-    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
-    ...theme.typography.body2,
-    padding: theme.spacing(1),
-    textAlign: "center",
-    color: theme.palette.text.secondary,
-}));
-
-//To be removed when global theme is done
-const buttonStyle = { color: COLORS.GREEN, border: `1px solid ${COLORS.GREEN}` };
+
+import {
+    getParticipants,
+    getParticipant,
+    getMeetings,
+} from "../../../../api/getRequests";
+import { spawnToast } from "../../../../utils/Toast";
+
+interface INITIALOWNER {
+    name: string;
+    id: number;
+}
 
 const QuickBook = () => {
-    const [timeButtonsVisible, setTimeButtonsVisible] = useState(false);
-    const [open, setOpen] = useState(false);
-    const [ownerName, setOwnerName] = useState("");
-    const [owners, setOwners] = useState([""]);
-    const [autoComplete, setAutoComplete] = useState(false);
+    const [timeButtonsVisible, setTimeButtonsVisible] =
+        useState<boolean>(false);
+    const [openQuickButtonMenu, setOpenQuickButtonMenu] =
+        useState<boolean>(false);
+    const [owner, setOwner] = useState<INITIALOWNER>();
+    const [possibleOwners, setPossibleOwners] = useState<string[]>([""]);
+    const [autoComplete, setAutoComplete] = useState<boolean>(false);
+    const [timeVal, setTimeVal] = useState<number>(0);
+    const [closestMeet, setClosestMeet] = useState<number>(0);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            let tempOwners: any[] = [];
+            let meetings_start_time: number[] = [];
+
+            const owners_response = await getParticipants();
+            owners_response.data.map(({ name }: { name: string }) =>
+                tempOwners.push(name)
+            );
```

Comment:
.map" returns a new array while ".forEach" return "undefined". So you have 2 options:

1.  Use ".map" and save the returned value:
`const owners = owners_response.data.map(({ name }: { name: string }) => {return name});`

2. Use ".forEach" and pus into a new array: 
`owners_response.data.forEach(({ name }: { name: string }) =>tempOwners.push(name) );`